src/Decompiler/Analysis/SsaState.cs

```diff
@@ -110,12 +110,67 @@ public void Dump(bool trace)
 			}
 		}
 
-		/// <summary>
-		/// Deletes a statement by removing all the ids it references 
-		/// from SSA state, then removes the statement itself from code.
-		/// </summary>
-		/// <param name="pstm"></param>
- 		public void DeleteStatement(Statement stm)
+        [Conditional("DEBUG")]
+        public void CheckUses()
+        {
+            CheckUses(s => Debug.WriteLine(s));
+        }
+
+        public void CheckUses(Action<string> error)
+        {
```

Comment:
OK, I'll create new class `InstructionUssCollector` to get actually used identifiers without modifying SSA state. And I'll use it instead of `InstructionUseAdder`/`InstructionUseRemover`